peptide_normalization.py

```diff
@@ -111,72 +124,96 @@ def intensity_normalization(dataset: DataFrame, field: str, class_field: str = "
+    # if class_field == "all":
+    #     normalize_df = dataset[[field]]
+    #     dataset[NORM_INTENSITY] = scaler.fit_transform(normalize_df)
+    # else:
+    #     # normalize taking into account samples
+    #     normalize_df = dataset[[PEPTIDE_SEQUENCE, CONDITION, field, class_field]]
+    #     # group peptide + charge into a single peptide intensity using the mean.
+    #     normalize_df = pd.pivot_table(normalize_df, values=field, index=[PEPTIDE_SEQUENCE, CONDITION],
+    #                                   columns=class_field,
+    #                                   aggfunc={field: np.mean})
+    #
+    #     print(normalize_df.head())
+    #
+    #     # Remove all peptides in less than 30% of the samples.
+    #     if remove_peptides:
+    #         normalize_df = remove_missing_values(normalize_df=normalize_df, ratio = 0.3)
+    #
+    #     # Imputation of the values using KNNImputer
+    #     # (https://scikit-learn.org/0.16/modules/generated/sklearn.preprocessing.Imputer.html)
+    #     if imputation:
+    #         imputer = SimpleImputer()
+    #         if imputation_method != "simple":
+    #             imputer = KNNImputer(n_neighbors=2, weights="uniform")
+    #         normalized_matrix = imputer.fit_transform(normalize_df)
+    #         if scaling_method != "quantile":
+    #             normalized_matrix = scaler.fit_transform(normalized_matrix)
+    #         else:
+    #             normalized_matrix = qnorm.quantile_normalize(normalized_matrix)
+    #     else:
+    #         normalized_matrix = scaler.fit_transform(normalize_df)
+    #
+    #     normalize_df[:] = normalized_matrix
+    #     normalize_df = normalize_df.reset_index()
+    #     normalize_df = normalize_df.melt(id_vars=[PEPTIDE_SEQUENCE, CONDITION])
+    #     normalize_df.rename(columns={'value': NORM_INTENSITY}, inplace=True)
+    #     dataset = pd.merge(dataset, normalize_df, how='left', on=[PEPTIDE_SEQUENCE, CONDITION, class_field])
+    #     dataset.pop(NORM_INTENSITY + "_x")
+    #     dataset.rename(
+    #         columns={NORM_INTENSITY + "_y": NORM_INTENSITY}, inplace=True)
 
-    # Scaler selection
-    elif scaling_method == "standard":
-        scaler = StandardScaler(with_mean=False)
-    elif scaling_method == "quantile":
-        scaler = QuantileTransformer(output_distribution="normal")
-    elif scaling_method == "normalizer":
-        scaler = Normalizer()
-    elif scaling_method == "robust":
-        scaler = RobustScaler()
-    elif scaling_method == "minmax":
-        scaler = MinMaxScaler()
-    elif scaling_method == 'maxabs':
-        scaler = MaxAbsScaler()
-
-    # normalize the intensities across all samples
-    if class_field == "all":
-        normalize_df = dataset[[field]]
-        dataset[NORM_INTENSITY] = scaler.fit_transform(normalize_df)
+    return dataset
+
+def get_peptidoform_normalize_intensities(dataset: DataFrame, higher_intensity = True) -> DataFrame:
```

Comment:
@jpfeuffer can you take a look to this function?

Reply:
Looks ok from here